### ISSUE TITLE
Twitter report: App crashes when importing a RAT via link click

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#5807]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Xiaomi 10t lite
- Android version: Android 12
- App version: 2.28.3

## Describe the bug

A Twitter user messaged me about his app crashing every time he tried to register a RAT via link click.

## Steps to reproduce the issue

1. Get tests with a rapid antigen test
2. Extract the link from the QR code or click on the link to import your result into the Corona-Warn-App
3. Notice that the app crashes

## Expected behaviour

The app should not crash. 

## Possible Fix

No idea. 

## Additional context

1. The user has captured his app crashing in an error report with the ID 068EDA6A362BD23B3205
2. The user also reported that when he tries to scan the QR code, his app throws the error reported in https://github.com/corona-warn-app/cwa-app-android/issues/5336
3. This issue is (obviously) based on feedback I received from a user via Twitter 

---
Internal Tracking ID: [EXPOSUREAPP-14610](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14610)

**Comment by Ein-Tim** [comment]
Will this be mirrored to JIRA? @dsarkar

**Comment by larswmh** [comment]
Thanks for forwarding this Twitter report @Ein-Tim. We have created an internal ticket for it and will raise this topic internally.
Internal Tracking ID: [EXPOSUREAPP-14610](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14610)

---
Corona-Warn-App Open Source Team

**Comment by mtwalli** [comment]
@Ein-Tim Thanks for reporting the issue, could you please ask the user what device is that and which Android version, The crash is not really related to the test, but rather to some style issue 

```
2023-01-18T16:51:11.726Z E/main: Uncaught exception!
java.lang.IllegalArgumentException: The style on this component requires your app theme to be Theme.AppCompat (or a descendant).
	at com.google.android.material.internal.ThemeEnforcement.checkTheme(ThemeEnforcement.java:40)
	at com.google.android.material.internal.ThemeEnforcement.checkCompatibleTheme(ThemeEnforcement.java:58)
	at com.google.android.material.internal.ThemeEnforcement.obtainStyledAttributes(ThemeEnforcement.java:1)
	at com.google.android.material.dialog.MaterialAlertDialogBuilder.<init>(MaterialAlertDialogBuilder.java:61)
	at kotlinx.coroutines.SupervisorKt.toErrorDialogBuilder(Supervisor.kt:8)
	at de.rki.coronawarnapp.ui.main.MainActivity$onCreate$12.invoke(MainActivity.kt:53)
	at de.rki.coronawarnapp.ui.main.MainActivity$$ExternalSyntheticLambda3.onChanged(R8$$SyntheticClass:10)
	at de.rki.coronawarnapp.util.ui.SingleLiveEvent$observe$1.invoke(SingleLiveEvent.kt:15)
	at de.rki.coronawarnapp.util.ui.SingleLiveEvent$$ExternalSyntheticLambda0.onChanged(R8$$SyntheticClass:8)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:30)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:16)
	at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:1)
	at de.rki.coronawarnapp.util.ui.SingleLiveEvent.setValue(SingleLiveEvent.kt:7)
	at androidx.lifecycle.LiveData$1.run(LiveData.java:19)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8292)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
```

**Comment by Ein-Tim** [comment]
@mtwalli Device and Android version are in the OP

**Comment by mtwalli** [comment]
@Ein-Tim I believe that this issue won't happen anymore in `v3.0.1` which is not fully rolled out. please ask the user to upgrade and check with the latest version

<img width="1037" alt="Bildschirm­foto 2023-01-19 um 17 18 39" src="https://user-images.githubusercontent.com/25054729/213496510-e34ecc93-9160-482f-8c75-174aff4fb523.png">

**Comment by Ein-Tim** [comment]
@mtwalli Thank you! And the error the user sees when importing the test (https://github.com/corona-warn-app/cwa-app-android/issues/5336) appears because the testing site has issued a QR code with corrupted data, right?

**Comment by mtwalli** [comment]
> @mtwalli Thank you! And the error the user sees when importing the test (https://github.com/corona-warn-app/cwa-app-android/issues/5336) appears because the testing site has issued a QR code with corrupted data, right? 

Correct 👍🏼

**Comment by Ein-Tim** [comment]
@mtwalli Indeed, with version 3.0 the app does not crash anymore and, what surprises me, the link even goes through, and the test is successfully registered, although when the user scans the QR code he sees the issue reported in https://github.com/corona-warn-app/cwa-app-android/issues/5336. Shouldn't clicking on the link lead to the same error? @thomasaugsten

**Comment by Ein-Tim** [comment]
- Follow-up issue #5814. 

This issue has been fixed. The `Fix 3.0` label can be applied to it. @larswmh

Closing as fixed.
